### ISSUE TITLE
Re-introduce App Insights request detail logging

### ISSUE THREAD

**Issue by brettsam** [Azure/azure-functions-host#2447]
When logging, the App Insights logger would look for an `HttpRequest` attached to the current logging scope. This request was attached by the Functions host. We'd use that to log more interesting request details like StatusCode, UserAgent, etc. In v2, this has changed as there's no guarantee that the Response has been written by the time the function execution has completed. That was leading to some tough scenarios.

In order to get tests running again, I've removed this behavior so we can take another look at how we scope the requests and the function invocations to see if we can get this to work more nicely.

**Comment by mathewc** [comment]
Pragna will link this to other related issues.

**Comment by pragnagopa** [comment]
Here is the related issue #2272 to add httprequest traces to systemlogs

**Comment by brettsam** [comment]
Note that this may require some refactoring of our HTTP pipeline. The problem is that we have no idea what the response code is when we log our 'Request' to App Insights. So while we may be able to log the ASP.NET logs, we won't be able to "merge" those with our function result. Not sure what the correct approach is here, other than making sure we've completed the Request before we actually signal that the function is complete.

**Comment by brettsam** [comment]
@lmolkova -- we can close this now that we track HTTP requests in the WebJobs/dev branch, right?

**Comment by lmolkova** [comment]
this is not addressed yet - I'll submit PR to webjobs when ApplicaitonInsights 2.8 is released (with distributed tracing support)

**Comment by MHHenriksen** [comment]
Looks like it's out! https://github.com/Microsoft/ApplicationInsights-dotnet-server/releases/tag/2.8.0

We're super eager to see this addressed.

**Comment by Squiggle** [comment]
Likewise. Do we have an ETA for this?

**Comment by brettsam** [comment]
We're reviewing a PR right now that will re-introduce this: #1965.

**Comment by MHHenriksen** [comment]
Great to see that the PR is merged! Are there any followups pending? Can you give an updated indication on the status of this issue, and rough time scale of next release?

**Comment by brettsam** [comment]
We had to revert this fix due to a few issues discovered during testing. We will be working to get this back in for our next release.

**Comment by MHHenriksen** [comment]
https://github.com/Azure/azure-webjobs-sdk/pull/2073 seems to be take two of fixing this.

**Comment by ishepherd** [comment]
It's not only Response details that are missing. We found things like URL (=> requests.url) and HTTP Method (=> requests.customDimensions.HttpMethod) are missing too.

**Comment by brettsam** [comment]
#2073 will address this as well. We are waiting on a portal change before we can fully merge the PR (it would break the query we use in the Functions "Monitor" page). It is effectively changing HTTP-triggered function telemetry to look like the standard App Insights RequestTelemetry.

**Comment by MHHenriksen** [comment]
@brettsam, did you link the right issue? #2073 was moved and then closed as wontfix, and doesn't seem related to what you wrote.

**Comment by bogardt** [comment]
nope, issue was moved to azure-functions-core-tools project from azure https://github.com/Azure/azure-functions-core-tools/issues/353

cf: https://github.com/Azure/azure-functions-host/issues/2073#issuecomment-358079463

**Comment by brettsam** [comment]
Whoops -- meant https://github.com/Azure/azure-webjobs-sdk/pull/2073.

**Comment by brettsam** [comment]
Bumping to Sprint 45. We're unblocked now but tracking down the last few bugs.

**Comment by fabiocav** [comment]
This is out in PR now, but moving to 46 as the work won't be completed until then.

**Comment by brettsam** [comment]
This was committed with #4214 and is being released with https://github.com/Azure/azure-functions-host/releases/tag/v2.0.12408
